src/pages/Main.test.ts

```diff
@@ -0,0 +1,210 @@
+    default: jest.fn().mockReturnValue(null),
+    __esModule: true,
+  };
+});
+
+type SupportRequestExports = typeof import('src/components/SupportRequest') & { __esModule: true };
+jest.mock('src/components/SupportRequest', (): SupportRequestExports => {
+  return {
+    ...jest.requireActual<SupportRequestExports>('src/components/SupportRequest'),
+    default: jest.fn().mockReturnValue(null),
+    __esModule: true,
+  };
+});
+
+type TitleManagerExports = typeof import('src/components/TitleManager');
+jest.mock('src/components/TitleManager', (): TitleManagerExports => {
+  return {
+    ...jest.requireActual<TitleManagerExports>('src/components/TitleManager'),
+    TitleManager: jest.fn().mockReturnValue(null),
+  };
+});
+
+type ImportStatusExports = typeof import('src/data/ImportStatus') & { __esModule: true };
+jest.mock('src/data/ImportStatus', (): ImportStatusExports => {
+  return {
+    ...jest.requireActual<ImportStatusExports>('src/data/ImportStatus'),
+    default: jest.fn().mockReturnValue(null),
+    __esModule: true,
+  };
+});
+
+type BrandUtilsExports = typeof import('src/libs/brand-utils');
+jest.mock('src/libs/brand-utils', (): BrandUtilsExports => {
+  const { defaultBrand } = jest.requireActual('src/libs/brands');
+  return {
+    ...jest.requireActual<BrandUtilsExports>('src/libs/brand-utils'),
+    getEnabledBrand: jest.fn().mockReturnValue(defaultBrand),
+    isBrand: jest.fn().mockImplementation((brand) => brand === defaultBrand),
+  };
+});
+
+type ErrorExports = typeof import('src/libs/error');
+jest.mock('src/libs/error', (): ErrorExports => {
+  return {
+    ...jest.requireActual<ErrorExports>('src/libs/error'),
+    reportError: jest.fn(),
+  };
+});
+
+type EventsExports = typeof import('src/libs/events');
+jest.mock('src/libs/events', (): EventsExports => {
+  return {
+    ...jest.requireActual<EventsExports>('src/libs/events'),
+    PageViewReporter: jest.fn().mockReturnValue(null),
+  };
+});
+
+type NavExports = typeof import('src/libs/nav');
+jest.mock('src/libs/nav', (): NavExports => {
+  return {
+    ...jest.requireActual<NavExports>('src/libs/nav'),
+    getCurrentUrl: jest.fn().mockReturnValue(''),
+    LocationProvider: jest.fn().mockImplementation(({ children }) => children),
+    PathHashInserter: jest.fn().mockReturnValue(null),
+    Router: jest.fn().mockReturnValue(null),
+  };
+});
+
+describe('Main', () => {
+  describe('theme', () => {
+    it('renders ThemeProvider with theme from enabled brand', () => {
+      // Arrange
+      const mockTheme: Theme = {} as Theme;
+      const mockBrandConfiguration: Partial<BrandConfiguration> = { theme: mockTheme };
+      asMockedFn(getEnabledBrand).mockReturnValue(mockBrandConfiguration as BrandConfiguration);
+
+      // Act
+      render(h(Main));
+
+      // Assert
+      expect(ThemeProvider).toHaveBeenCalledWith(expect.objectContaining({ theme: mockTheme }), expect.anything());
```

Comment:
whenever I use objectContaining, I am sure to have type safety on the object I am giving it, 'satisfies' works great for this.

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/4247/commits/c4453a5217826135728b1481ce1ab526313f47a5. Had to export the type from components.